src/Worker/Core/Shims/TaskOrchestrationContextWrapper.cs

```diff
@@ -14,6 +14,7 @@ namespace Microsoft.DurableTask.Worker.Shims;
 /// </summary>
 sealed partial class TaskOrchestrationContextWrapper : TaskOrchestrationContext
 {
+    // TODO: Change externalEventSources value from IEventSource to queue<IEventSource> to preserve FIFO in O(1)
```

Comment:
I'd prefer we do this as part of this PR since I'm a little worried about the implications of making this O(N). Is that something we can consider?

Reply:
100% I'll get that added in

Reply:
I would go 2 steps further and mark [`IEventSource.Next`](https://github.com/microsoft/durabletask-dotnet/blob/e2f329258d1e02a762911bd89274a9a584b6ec04/src/Worker/Core/Shims/TaskOrchestrationContextWrapper.EventSource.cs#L17) as deprecated. It is only used in this project for this \$O(n)\$ FIFO implementation.  Using a Queue will make this code simpler to read as it would make the intent of this map explicit rather then implicit.

The only other place in this codebase it is used is at the event completion:

https://github.com/microsoft/durabletask-dotnet/blob/e2f329258d1e02a762911bd89274a9a584b6ec04/src/Worker/Core/Shims/TaskOrchestrationContextWrapper.cs#L314-L322

I don't know if it is used externally, hence why marking it as deprecated is the way to go if this is a stable API.

Reply:
`IEventSource` is a private internal class, so we don't have to worry about making breaking changes to it. We can just go ahead and remove the `Next` property.